File: examples/android/MSDKSampleAppJava/app/src/main/java/msdk/java/sample/MainPagerAdapter.java
package msdk.java.sample;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import msdk.java.sample.history.HistoryFragment;
import msdk.java.sample.homepage.HomePageFragment;

public class MainPagerAdapter extends FragmentPagerAdapter {
    public MainPagerAdapter(@NonNull FragmentManager fm) {
        super(fm);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return HomePageFragment.newInstance();
            case 1:
                return HistoryFragment.newInstance();
            default:
                return null;
        }
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return "Home";
            case 1:
                return "History";
            default:
                return null;
        }
    }


    @Override
    public int getCount() {
        return 2;
    }
}

File: examples/android/MSDKSampleAppKotlin/app/src/main/java/msdk/kotlin/sample/MainPagerAdapter.kt
package msdk.kotlin.sample

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import msdk.kotlin.sample.history.HistoryFragment
import msdk.kotlin.sample.homepage.HomePageFragment


class MainPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
    override fun getItem(position: Int): Fragment = when (position) {
        0 -> HomePageFragment.newInstance()
        1 -> HistoryFragment.newInstance()
        else -> HomePageFragment.newInstance()
    }


    override fun getPageTitle(position: Int): CharSequence? = when (position) {
        0 -> "Home"
        1 -> "History"
        else -> null
    }

    override fun getCount(): Int = 2
}
